cartridge/shop/views.py

```diff
@@ -282,6 +299,15 @@ def checkout_steps(request, form_class=OrderForm, extra_context=None):
                 # and send the order receipt email.
                 order = form.save(commit=False)
                 order.setup(request)
+
+                if (not settings.SHOP_PAYMENT_STEP_ENABLED or
+                        order.total == 0):
```

Comment:
step += 1 for some reason doesn't completely skip the payment. You can try with the master branch, activate a payment provider (like stripe) and `SHOP_PAYMENT_STEP_ENABLED = False`, you'll get an error from the payment processor...

By adding this line I just use the dummy payment processor when the order is zero.